### PR TITLE
feat(@clayui/core): make it possible to enable or disable drag and drop

### PR DESCRIPTION
**Pull request by julien**
See #4389

### CODE REVIEW

File: packages/clay-core/src/tree-view/TreeView.tsx [hunk 1]
```diff
@@ -21,6 +21,7 @@ interface ITreeViewProps<T>
 		ITreeProps<T>,
 		ICollectionProps<T> {
 	displayType?: 'light' | 'dark';
+	dragAndDropEnabled?: boolean;
```
[the remark this answers is not in the document]
Comment (julien) [reply packages/clay-core/src/tree-view/TreeView.tsx:24]: Yes I agree, you know naming is hard - `dragAndDrop` seems good to me, will update

File: packages/clay-core/src/tree-view/useItem.tsx [hunk 2]
```diff
@@ -80,6 +80,9 @@ export function ItemContextProvider({children, value}: Props) {
 	const [overPosition, setOverPosition] = useState<Position | null>(null);
 
 	const [{isDragging}, drag, preview] = useDrag({
+		canDrag() {
+			return dragAndDropEnabled ?? false;
+		},
```
[the remark this answers is not in the document]
Comment (julien) [reply packages/clay-core/src/tree-view/useItem.tsx:85]: Oh yeah, that makes sense. You mean something like 

```suggestion
canDrag() {
    return items && items.length && dragAndDrop ?? false;
},
```

File: packages/clay-core/src/tree-view/TreeView.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (matuzalemsteles) [reply packages/clay-core/src/tree-view/TreeView.tsx:24]: Yes, don't worry I'm terrible with names 😅

File: packages/clay-core/src/tree-view/useItem.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (matuzalemsteles) [reply packages/clay-core/src/tree-view/useItem.tsx:85]: Oh sorry, I was referring to this line 😅https://github.com/liferay/clay/blob/536706168e53b427d95fce5dacf18467f6f005ab/packages/clay-core/src/tree-view/useItem.tsx#L186

[the remark this answers is not in the document]
Comment (julien) [reply packages/clay-core/src/tree-view/useItem.tsx:85]: Gotcha :+1: Seems good to me

**Review by matuzalemsteles — APPROVED** [review]
LGTM

**Pull request merged**